ambry-router/src/main/java/com/github/ambry/router/GetBlobInfoOperation.java

```diff
@@ -300,6 +300,7 @@ void handleResponse(ResponseInfo responseInfo, GetResponse getResponse) {
   private void processGetBlobInfoResponse(RequestInfo getRequestInfo, GetResponse getResponse)
       throws IOException, MessageFormatException {
     ServerErrorCode getError = getResponse.getError();
+    responseHandler.onEvent(getRequestInfo.getReplicaId(), getError);
```

Comment:
Just wondering if we might be notifying response handler of server or partition error in [RouterUtils.extractResponseAndNotifyResponseHandler](https://github.com/linkedin/ambry/blob/1339c995283f00555173f820ed22e47ab15fbf1b/ambry-router/src/main/java/com/github/ambry/router/RouterUtils.java#L226).

Reply:
That's the line for Response's ServerErrorCode. For example. DeleteResponse has a ServerErrorCode. After we deserialized a DeleteResponse, we will notify ResponseHandler with DeleteResponse.getError. The logic applies to all other Responses.
However, GetResponse is different, because it carries multiple layer of errors. First we have a GetResponse.getError, and for each partition, we have another error https://github.com/linkedin/ambry/blob/master/ambry-protocol/src/main/java/com/github/ambry/protocol/PartitionResponseInfo.java#L38. 
An in server, we already return an NoError in GetResponse. The real error for each partition is buried in the PartitionResponseInfo.

Reply:
Hmm, may be we might be taking care of error from PartitionResponseInfo as well by we call `RouterUtils.extractResponseAndNotifyResponseHandler` from [GetManager.handleResponse](https://github.com/linkedin/ambry/blob/1339c995283f00555173f820ed22e47ab15fbf1b/ambry-router/src/main/java/com/github/ambry/router/GetManager.java#L226)?

Reply:
that's right, then I can close this one.